lib/internal/crypto/cfrg.js

```diff
@@ -329,18 +329,21 @@ function cfrgImportKey(
     extractable);
 }
 
-function eddsaSignVerify(key, data, { name, context }, signature) {
+function validateEdDSASignVerifyAlgorithm(algorithm) {
+  if (algorithm.name === 'Ed448' && algorithm.context?.byteLength) {
+    throw lazyDOMException(
+      'Non zero-length context is not yet supported.', 'NotSupportedError');
+  }
+}
+
+function eddsaSignVerify(key, data, algorithm, signature) {
+  validateEdDSASignVerifyAlgorithm(algorithm);
```

Comment:
This changes the order of the checks. Previously, we first checked `(key.type !== type)` and then checked `(name === 'Ed448' && context?.byteLength)`. Now, it's the other way around.

It seems like we're not really following the strict order [from the specification](https://www.w3.org/TR/WebCryptoAPI/) for other algorithms either. So I guess this is fine?

Reply:
Yeah, the `(name === 'Ed448' && context?.byteLength)` check is our own because we don't support non-empty contexts. From the (early draft warning) [supports method proposal](https://twiss.github.io/webcrypto-modern-algos/#partial-subtlecrypto-interface-checking-support-for-algorithm) this falls under 

> If the specified operation or algorithm (or one of its parameter values) is expected to fail (for any key and/or data) for an implementation-specific reason (e.g. known nonconformance to the specification), return false.

Reply:
I have a draft PR for the supports method in #57270 that I will further refactor this PR's validations structures in when it lands.